### ISSUE TITLE
[Bug]: Traefik https redirect not working

### ISSUE THREAD

**Issue by majkshkurti** [kube-hetzner/terraform-hcloud-kube-hetzner#1242]
### Description

Hello. 
I tried setting up a cluster and I noticed that the traefik https redirect doesn't seem to work. Maybe I am doing something wrong but this was also pointed out here in a discussion topic so I thought it might be related. 
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/1133
I did some investigation and I think that the changes related to this issue is a traefik redirect "priority" value from this commit. 
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/commit/911730d7828629afebc0745c92788c4344e6885a
If I remove the `--entrypoints.web.http.redirections.entryPoint.priority=10`  it seems to work again. 

### Kube.tf file

```terraform
module "kube-hetzner" {
  providers = {
    hcloud = hcloud
  }
  hcloud_token    = var.hcloud_token
  source          = "kube-hetzner/kube-hetzner/hcloud"
  ssh_public_key  = file(var.kube_hetzner_ssh_public_key)
  ssh_private_key = file(var.kube_hetzner_ssh_private_key)
  network_region  = "eu-central"
  cluster_name    = "dev"
  control_plane_nodepools = [
    {
      name        = "control-plane-nbg1",
      server_type = "cpx11",
      location    = "nbg1",
      labels      = [],
      taints      = [],
      count       = 1
    },
  ]
  agent_nodepools = [
    {
      name        = "agent-large",
      server_type = "cpx31",
      location    = "nbg1",
      labels      = [],
      taints      = [],
      count       = 1
    }
  ]
  load_balancer_type       = "lb11"
  load_balancer_location   = "nbg1"
  ssh_max_auth_tries       = 15
  automatically_upgrade_os = false
  create_kubeconfig        = false
  kured_version            = "1.13.1"
}
```

### Screenshots

![image](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/32641116/abce1bd3-51bf-4de3-ab57-6337cef72c4d)

### Platform

Linux

**Comment by mysticaltech** [comment]
@majkshkurti Thanks for the debugging. PR most welcome please.

**Comment by mysticaltech** [comment]
@majkshkurti The fix was shipped in v2.13.0.

**Comment by majkshkurti** [comment]
Thank you @mysticaltech
